Gifski/ConversionCompletedViewController.swift

```diff
@@ -0,0 +1,159 @@
+import Cocoa
+import Quartz
+import UserNotifications
+import StoreKit
+
+final class ConversionCompletedViewController: NSViewController {
+	@IBOutlet private var draggableFileWrapper: NSView!
+	@IBOutlet private var fileNameLabel: Label!
+	@IBOutlet private var fileSizeLabel: Label!
+	@IBOutlet private var saveAsButton: NSButton!
+	@IBOutlet private var shareButton: NSButton!
+	@IBOutlet private var copyButton: NSButton!
+
+	private let draggableFile = DraggableFile()
+	private var conversion: Gifski.Conversion!
+	private var gifUrl: URL!
+
+	convenience init(conversion: Gifski.Conversion, gifUrl: URL) {
+		self.init()
+
+		self.conversion = conversion
+		self.gifUrl = gifUrl
+	}
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		setupUI()
+		setupDropView()
+		setup(url: gifUrl)
+	}
+
+	override func viewDidAppear() {
+		super.viewDidAppear()
+
+		if #available(macOS 10.14, *), defaults[.successfulConversionsCount] == 5 {
+			SKStoreReviewController.requestReview()
+		}
+
+		if #available(macOS 10.14, *), !NSApp.isActive || view.window?.isVisible == false {
+			let notification = UNMutableNotificationContent()
+			notification.title = "Conversion Completed"
+			notification.subtitle = conversion.video.filename
+			let request = UNNotificationRequest(identifier: "conversionCompleted", content: notification, trigger: nil)
+			UNUserNotificationCenter.current().add(request)
+		}
+
+		draggableFile.layer?.animateScaleMove(fromScale: 3, fromY: view.frame.height + draggableFile.frame.size.height)
+	}
+
+	private func setupUI() {
```

Comment:
```suggestion
	private func setUpUI() {
```

Reply:
hmm, in the app the convention is always `setup` rather than `setUp` - do you prefer that I change all the places to `setUp` instead?

Reply:
Because alone it can be used as a noun `setup`, but when you have `UI` afterwards it needs to be a verb `set up`. It could be `uiSetup`.

https://writingexplained.org/setup-vs-set-up-difference